### PR TITLE
feat(DEV-10046): migration from SDK API to Qraft in `<TagsTable />` 🏷️

### PR DESCRIPTION
**Pull request by radist2s**
This PR focuses on the `<TagsTable />` component, migrating it from the SDK API to Qraft.

**Key changes within `<TagsTable />`:**

* **Qraft integration:**
    * Replaced SDK API calls with Qraft for all API requests _(v2024-05-25)_ within the component.
    * Updated dependencies and imports accordingly.
*  **Direct Qraft Hook Usage:**
    * 🆕 Removed custom `useTag*()` hooks.
    * 🆕 Implemented direct usage of Qraft hooks, simplifying type maintenance and moving error handling (toast calls) closer to the component code.
* **Auth Demo:**
    * Updated demo data generator API usage up to _v2024-05-25_

### CODE REVIEW

File: packages/sdk-react/package.json [hunk 1]
```diff
@@ -23,7 +24,11 @@
     "compile-translations": "lingui compile",
     "write-package-version-file": "yarn exec ../../write-package-version-file.sh",
     "storybook": "storybook dev -p 6006",
-    "build-storybook": "storybook build -o storybook_static"
+    "build-storybook": "storybook build -o storybook_static",
+    "download-openapi-document": "curl -o 'src/api/schema.json' \"https://api.dev.monite.com/openapi.json?version=$(node -p 'require(\"@monite/sdk-react/package.json\").apiVersion')\" && prettier --write src/api/schema.json",
```
Comment (saladNights) [inline packages/sdk-react/package.json:28]: Should the link to the API contain a dynamic stand? Or at least let's get the schema from the sandbox. dev doesn't look like a reliable source

File: packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.tsx [hunk 2]
```diff
@@ -158,7 +140,11 @@ const ProductEditFormBase = (props: IProductEditFormProps) => {
       description: values.description,
     };
 
-    return updateProduct(payload);
+    return productUpdateMutation.mutate(payload, {
+      onSuccess: (product) => {
+        props.onUpdated?.(product);
```
Comment (saladNights) [inline packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.tsx:145]: in the previous version of the onSuccess callback there was `props.onCanceled();`.
It changes the view mode of the details dialog, from edit to read.
In your current implementation, the form persists after changes are successfully saved.

We should either put `onCancel` back, or introduce another, more explicit variable (`viewMode`, `isEdit`, etc)

File: packages/sdk-react/src/core/queries/useTag.ts [hunk 3]
```diff
@@ -1,79 +1,90 @@
...
-    { id: string; payload: { name: string } }
-  >({
-    mutationFn: ({ id, payload }) => monite.api.tag.update(id, payload),
-
-    onSuccess: () => {
-      invalidate();
-    },
+export const useCreateTagMutation = () => {
+  const { i18n } = useLingui();
+  const { api, apiVersion, entityId } = useMoniteApiClient();
```
Comment (saladNights) [inline packages/sdk-react/src/core/queries/useTag.ts:23]: unused variables `apiVersion`, `entityId`

File: packages/sdk-react/src/components/tags/TagFormModal/TagFormModal.tsx [hunk 4]
```diff
@@ -90,7 +95,11 @@ const TagFormModalBase = ({
     const tagCreateMutate = tagCreateMutation.mutate;
     tagCreateMutate(
       {
-        name,
+        header: {
```
Comment (saladNights) [inline packages/sdk-react/src/components/tags/TagFormModal/TagFormModal.tsx:98]: sorry, I don't get it why should we pass headers here? Can we define headers like in `useUpdateTagMutation`?

**Review by saladNights — CHANGES_REQUESTED** [review]

File: packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.tsx (hunk 2 above)
Comment (radist2s) [reply packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.tsx:145]: Thanks! I was expecting someone familiar with this to come along and explain this behavior to me. Don't you mind if I add the `todo` and advice for the refactoring in the code?

File: packages/sdk-react/src/core/queries/useTag.ts (hunk 3 above)
Comment (radist2s) [reply packages/sdk-react/src/core/queries/useTag.ts:23]: Thank you, I even refactored this mutation hook.

File: packages/sdk-react/src/components/tags/TagFormModal/TagFormModal.tsx (hunk 4 above)
Comment (radist2s) [reply packages/sdk-react/src/components/tags/TagFormModal/TagFormModal.tsx]: Nice advice, thank you! Refactored.

File: packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.tsx (hunk 2 above)
Comment (saladNights) [reply packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.tsx]: yes, good for me

**Review by nekedos — CHANGES_REQUESTED** [review]

File: packages/sdk-react/src/core/context/QraftProvider.tsx
Comment (nekedos) [inline packages/sdk-react/src/core/context/QraftProvider.tsx:1]: Don't you think renaming this file to `MoniteAPIProvider` would be better?

Comment (radist2s) [reply packages/sdk-react/src/core/context/QraftProvider.tsx:1]: That was the idea, thanks!

**Review by nekedos — APPROVED** [review]

File: packages/sdk-react/package.json [hunk 5]
```diff
@@ -23,7 +24,11 @@
...
+    "codegen": "yarn codegen:api-version && yarn codegen:generate-api-client",
+    "codegen:generate-api-client": "openapi-qraft --plugin tanstack-query-react --plugin openapi-typescript --output-dir src/api --filter-services '**,!/internal/**' src/api/schema.json && prettier --write src/api/**/**.ts",
```
[the remark this answers is not in the document]
Comment (radist2s) [reply packages/sdk-react/package.json]: Done

**Pull request merged**